src/renderer/components/test-utils/get-application-builder.tsx

```diff
@@ -62,34 +57,50 @@ import activeKubernetesClusterInjectable from "../../cluster-frame-context/activ
 import { catalogEntityFromCluster } from "../../../main/cluster-manager";
 import namespaceStoreInjectable from "../+namespaces/store.injectable";
 import { isAllowedResource } from "../../../common/cluster/is-allowed-resource";
+import createApplicationWindowInjectable from "../../../main/start-main-application/lens-window/application-window/create-application-window.injectable";
+import type { CreateElectronWindow } from "../../../main/start-main-application/lens-window/application-window/create-electron-window.injectable";
+import createElectronWindowInjectable from "../../../main/start-main-application/lens-window/application-window/create-electron-window.injectable";
+import { applicationWindowInjectionToken } from "../../../main/start-main-application/lens-window/application-window/application-window-injection-token";
+import sendToChannelInElectronBrowserWindowInjectable from "../../../main/start-main-application/lens-window/application-window/send-to-channel-in-electron-browser-window.injectable";
+import closeAllWindowsInjectable from "../../../main/start-main-application/lens-window/hide-all-windows/close-all-windows.injectable";
+import type { LensWindow } from "../../../main/start-main-application/lens-window/application-window/create-lens-window.injectable";
+import type { FakeExtensionOptions } from "./get-extension-fake";
+import { getExtensionFakeForMain, getExtensionFakeForRenderer } from "./get-extension-fake";
 
-type Callback = (dis: DiContainers) => void | Promise<void>;
+type Callback = (di: DiContainer) => void | Promise<void>;
 
-type EnableExtensions<T> = (...extensions: T[]) => void;
-type DisableExtensions<T> = (...extensions: T[]) => void;
+type LensWindowWithHelpers = LensWindow & { rendered: RenderResult; di: DiContainer };
 
 export interface ApplicationBuilder {
-  dis: DiContainers;
+  mainDi: DiContainer;
   setEnvironmentToClusterFrame: () => ApplicationBuilder;
 
   extensions: {
-    renderer: {
-      enable: EnableExtensions<LensRendererExtension>;
-      disable: DisableExtensions<LensRendererExtension>;
-    };
+    enable: (...extensions: FakeExtensionOptions[]) => void;
+    disable: (...extensions: FakeExtensionOptions[]) => void;
+
+    get: (id: string) => {
+      main: LensMainExtension;
 
-    main: {
-      enable: EnableExtensions<LensMainExtension>;
-      disable: DisableExtensions<LensMainExtension>;
+      applicationWindows: Record<string, LensRendererExtension> & {
+        only: LensRendererExtension;
+      };
     };
```

Comment:
In 99,9% cases we unit test situations where there is only one application window. That's why I introduced these `only` getters which can be utilized so that you don't have to specify the ID. 

There is one instance of LensRendererExtension for each window, so that's why we need something like this.